lib/model/kademliasql.ex

```diff
@@ -115,20 +141,56 @@ defmodule Model.KademliaSql do
     end
   end
 
+  defp decode_binary_row([key, object_blob, stored_at]) do
+    decode_binary(key, object_blob, stored_at)
+  end
+
+  defp decode_term_row(row) do
+    {key, binary, stored_at} = decode_binary_row(row)
+    {key, Object.decode!(binary), stored_at}
+  end
+
+  defp decode_binary(key, object_blob, stored_at) do
+    {key, BertInt.decode!(object_blob), stored_at}
+  end
+
+  def prune_stale_objects() do
+    cutoff = now_seconds() - @stale_prune_seconds
+
+    stale_keys =
+      query!("SELECT key FROM p2p_objects WHERE stored_at < ?1", [cutoff])
+      |> Enum.map(&hd/1)
+
+    if stale_keys != [] do
+      query!("DELETE FROM p2p_objects WHERE stored_at < ?1", [cutoff])
+      KademliaLight.drop_nodes(stale_keys)
+    end
```

Comment:
1. Add a `LIMIT 100` statement and if `length(state_keys) > 0`
2. Run KademliaLight.drop_nodes before the DELETE statement